datajunction-server/alembic/versions/2023_10_23_2051-363991e344b8_add_created_by_for_nodes_namespaces_and_.py

```diff
@@ -0,0 +1,61 @@
+"""Add created_by for nodes, namespaces, and tags
+
+Revision ID: 363991e344b8
+Revises: 58e246fba34b
+Create Date: 2023-10-23 20:51:40.076102+00:00
+
+"""
+# pylint: disable=no-member, invalid-name, missing-function-docstring, unused-import, no-name-in-module
+
+import sqlalchemy as sa
+import sqlmodel
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "363991e344b8"
+down_revision = "58e246fba34b"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
```

Comment:
Should this migration include a stage that will populate the existing nodes' `created_by` fields with the username extracted from history events? That would make the info more complete where possible

Reply:
+1. I didn't realize that steps like that could be included in the migration itself. But this makes total sense, because this step could just be a SQL update statement.

Reply:
That's a great idea, let me try that out 👀

Reply:
@shangyian take a look when you have a chance. I insert a new user `"unknown"` as part of the upgrade and backfill the `created_by` values everywhere with that user ID. Is this what you imagined?

Reply:
@samredai this looks good to me! I guess another option is we could look at each node's history events and extract the user from there, but that might be too smart, since that user won't ever be able to log in.

Reply:
Ah right, I'm re-reading your first comment where you mentioned using the history events. Good point about how they wouldn't be able to login. Hopefully this is an easy cleanup of existing nodes where system admins can just adhoc reassign all of these "unknown" users based on the node history and then new creations are properly assigned.

Reply:
Looks good imo. And we should be ok for now, but I think we should make sure our integration test captures some of the actions that are affected here.